src/utils/storage.ts

```diff
@@ -393,3 +412,254 @@ function encodeVariable(
+ * @param variableName Variable name to find against the given storage layout.
+ * @param vmManager SmockVMManager is used to get certain storage values given a specific slot key and a contract address
+ * @param contractAddress Contract address to use for vmManager
+ * @param mappingKey Only used for mappings, represents they key of a mapping value
+ * @returns An array of storage slot key/type pair that would result in the value of the variable.
+ */
+export async function getVariableStorageSlots(
+  storageLayout: SolidityStorageLayout,
+  variableName: string,
+  vmManager: SmockVMManager,
+  contractAddress: string,
+  mappingKey?: string | number
+): Promise<StorageSlotKeyTypePair[]> {
+  // Find the entry in the storage layout that corresponds to this variable name.
+  const storageObj = storageLayout.storage.find((entry) => {
+    return entry.label === variableName;
+  });
+
+  // Complain very loudly if attempting to set a variable that doesn't exist within this layout.
+  if (!storageObj) {
+    throw new Error(`Variable name not found in storage layout: ${variableName}`);
+  }
+
+  const storageObjectType: SolidityStorageType = storageLayout.types[storageObj.type];
+
+  // Here we will store all the key/type pairs that we need to get the variable's value
+  let slotKeysTypes: StorageSlotKeyTypePair[] = [];
+  let key: string =
+    '0x' +
+    remove0x(
+      BigNumber.from(0)
+        .add(BigNumber.from(parseInt(storageObj.slot, 10)))
+        .toHexString()
+    ).padStart(64, '0');
+
+  if (storageObjectType.encoding === 'inplace') {
+    // For `inplace` encoding we only need to be aware of structs where they take more slots to store a variable
+    if (storageObjectType.label.startsWith('struct')) {
+      if (storageObjectType.members === undefined) {
+        throw new Error(`There are no members in object type ${storageObjectType}`);
+      }
+      // Slot key that represents the struct
+      slotKeysTypes = slotKeysTypes.concat({
+        key: key,
+        type: storageObjectType,
+        offset: storageObj.offset,
+      });
+      // These slots are for the members of the struct
+      for (let i = 0; i < storageObjectType.members.length; i++) {
+        // We calculate the slot key for each member
+        key = '0x' + remove0x(BigNumber.from(key).add(BigNumber.from(storageObjectType.members[i].slot)).toHexString()).padStart(64, '0');
+        slotKeysTypes = slotKeysTypes.concat({
+          key: key,
+          type: storageLayout.types[storageObjectType.members[i].type],
+          label: storageObjectType.members[i].label,
+          offset: storageObjectType.members[i].offset,
+        });
+      }
+    } else {
+      // In cases we deal with other types than structs we already know the slot key and type
+      slotKeysTypes = slotKeysTypes.concat({
+        key: key,
+        type: storageObjectType,
+        offset: storageObj.offset,
+      });
```

Comment:
i know that this is out of scope for this PR, but please check #118 
should this be affected the issue?

Reply:
The `setVariable` bug doesn't affect this PR